libraries/AP_Terrain/AP_Terrain.cpp

```diff
@@ -309,12 +304,9 @@ void AP_Terrain::update(void)
     // just schedule any needed disk IO
     schedule_disk_io();
 
-    // try to ensure the home location is populated
-    float height;
-    height_amsl(ahrs.get_home(), height, false);
```

Comment:
As far as I can tell by the time we ever need height above home we have already checked in that location. Relying on update() to cache it when we already will manage this later once we needed it just seemed excessive.

Part of the fix was about squashing unread return values, but by inspection I was struggling to see any argument to actually call it here as it will normally just be noise on call time.